### PR TITLE
add support for ticket-based filtering of events

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/115583

Based on
* https://github.com/moderntribe/tribe-common/pull/798
* https://github.com/moderntribe/the-events-calendar/pull/2227

This PR adds a decorator of the base event repository to provide ticket-powered filtering of events.  
The code covers ET+ integration too.

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Comma required after last value in array declaration | [Event_Repository.php:59](https://github.com/moderntribe/event-tickets/blob/4e8f127bfc1e5744e9aa52a6af5958f2c04d45ec/src/Tribe/Event_Repository.php#L59)
:x: | Comma required after last value in array declaration | [Event_Repository.php:66](https://github.com/moderntribe/event-tickets/blob/4e8f127bfc1e5744e9aa52a6af5958f2c04d45ec/src/Tribe/Event_Repository.php#L66)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Comma required after last value in array declaration | [Event_Repository.php:59](https://github.com/moderntribe/event-tickets/blob/b2abe84d194280394d4488f0cdaa435fa2df21a2/src/Tribe/Event_Repository.php#L59)
:x: | Comma required after last value in array declaration | [Event_Repository.php:66](https://github.com/moderntribe/event-tickets/blob/b2abe84d194280394d4488f0cdaa435fa2df21a2/src/Tribe/Event_Repository.php#L66)

> via jenkins/codesniffer-phpcs

File: tests/wpunit/Tribe/Tickets/Event_RepositoryTest.php [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+	protected $events = [];
+
+	public function setUp() {
+		// before
+		parent::setUp();
+
+		// your set up methods here
+		$this->factory()->event = new Event();
+	}
+
```
Comment (sc0ttkclark) [inline tests/wpunit/Tribe/Tickets/Event_RepositoryTest.php]: Can you add at least one test that confirms one of the original Event Repo 'where' works as expected?

Comment (lucatume) [reply tests/wpunit/Tribe/Tickets/Event_RepositoryTest.php]: I did in the latest commit.

**Comment by sc0ttkclark** [comment]
Test failing still, but ready for merge when that's resolved

**Pull request merged**